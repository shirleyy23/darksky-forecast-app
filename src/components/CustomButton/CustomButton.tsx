import React from 'react';
import {
  Link as RouterLink,
  LinkProps as RouterLinkProps,
} from 'react-router-dom';
import Button from '@material-ui/core/Button';
import { Omit } from '@material-ui/types';
import { makeStyles } from '@material-ui/core/styles';

interface Props {
  link: string;
  content: string;
}

const useStyles = makeStyles({
  button: {
    padding: '.75em 2.25em',
    borderRadius: '30px',
    fontWeight: 800,
  },
});

const LinkBehavior = React.forwardRef<any, Omit<RouterLinkProps, 'to'>>(
  (props, ref) => (
    <RouterLink ref={ref} to="/getting-started/installation/" {...props} />
  )
);

LinkBehavior.displayName = 'Link Behaviour';

const CustomButton: React.FC<Props> = ({ content, link }) => {
  const classes = useStyles();
  return (
    <Button
      variant="contained"
      color={link === '/' ? 'secondary' : 'primary'}
      component={RouterLink}
      className={classes.button}
      to={link}
    >
      {content}
    </Button>
  );
};
export default CustomButton;
