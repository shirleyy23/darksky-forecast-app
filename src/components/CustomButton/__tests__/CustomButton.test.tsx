import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import CustomButton from '../CustomButton';

type CustomButtonProps = React.ComponentProps<typeof CustomButton>;

const renderCustomButton = (props: Partial<CustomButtonProps> = {}) => {
  const defaultProps: CustomButtonProps = {
    content: 'SEARCH',
    link: '/forecast',
  };
  return render(
    <BrowserRouter>
      <CustomButton {...defaultProps} {...props} />
    </BrowserRouter>
  );
};

describe('<CustomButton />', () => {
  test('<CustomButton /> renders with title and link', () => {
    const { container } = renderCustomButton();
    expect(container.firstChild).toMatchSnapshot();
  });
});
