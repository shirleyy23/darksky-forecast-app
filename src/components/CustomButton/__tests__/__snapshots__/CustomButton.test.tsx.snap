// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomButton /> <CustomButton /> renders with title and link 1`] = `
<a
  aria-disabled="false"
  class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1 MuiButton-containedPrimary"
  data-testid="link"
  href="/forecast"
  role="button"
  tabindex="0"
>
  <span
    class="MuiButton-label"
  >
    SEARCH
  </span>
  <span
    class="MuiTouchRipple-root"
  />
</a>
`;
